### PR TITLE
Making hrp2-torque-control and sot-torque-control independent.

### PR DESCRIPTION
**Pull request by olivier-stasse**
This new pull request is rebased on latest modification of sot-v3.

### CODE REVIEW

File: src/common-py.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    .def("set_force_id_right_foot", &ForceUtil::set_force_id_right_foot)
+    .def("display",                 &ForceUtil::display)
+    ;
+
+  class_<RobotUtil>("RobotUtil")
+    .def_readwrite("m_force_util",&RobotUtil::m_force_util)
+    .def_readwrite("m_foot_util",&RobotUtil::m_foot_util)
+    .def_readwrite("m_urdf_to_sot",&RobotUtil::m_urdf_to_sot)
+    .def("set_joint_limits_for_id",&RobotUtil::set_joint_limits_for_id)
+    .def("get_joint_limits_from_id",&RobotUtil::cp_get_joint_limits_from_id)
```
Comment (andreadelprete) [inline src/common-py.cpp]: Why is the method called ```cp_get_joint_limits_from_id```? I'm referring to the initial ```cp```.

File: src/control-manager.cpp [hunk 2]
```diff
@@ -112,15 +111,107 @@ namespace dynamicgraph
...
+	m_maxCurrent = lmax_current;
+
         m_dt = dt;
         m_emergency_stop_triggered = false; 
         m_initSucceeded = true;
+	vector<string> package_dirs;
+
+	m_robot = new pininvdyn::RobotWrapper(urdfFile, package_dirs, se3::JointModelFreeFlyer());
+	
+	std::string localName("control-manager-robot");
```
Comment (andreadelprete) [inline src/control-manager.cpp]: I know I already said it, but just as a reminder, this string "control-manager-robot" should be taken as an input parameter of the init command.

File: src/free-flyer-locator.cpp [hunk 3]
```diff
@@ -65,43 +63,75 @@ namespace dynamicgraph
...
+	      return;
+	    }
+
+	  m_model = new se3::Model();
+	  m_model->name.assign("EmptyRobot");
+
+          se3::urdf::buildModel(m_robot_util->m_urdf_filename,
+				se3::JointModelFreeFlyer(),*m_model);
+	  assert(m_model->nv == m_robot_util->m_nbJoints+6);
+          assert(m_model->existFrame(m_robot_util->m_foot_util.m_Left_Foot_Frame_Name));
```
Comment (andreadelprete) [inline src/free-flyer-locator.cpp:115]: Are you sure that ```m_Left_Foot_Frame_Name``` has been set at this point?

File: src/free-flyer-locator.cpp [hunk 4]
```diff
@@ -155,12 +185,12 @@ namespace dynamicgraph
...
           // due to distance from ankle to ground
-          Eigen::Map<const Eigen::Vector3d> righ_foot_sole_xyz(&RIGHT_FOOT_SOLE_XYZ[0]);
+          Eigen::Map<const Eigen::Vector3d> righ_foot_sole_xyz(&m_robot_util->m_foot_util.m_Right_Foot_Sole_XYZ[0]);
 
-          m_q_sot.tail<N_JOINTS>() = q.tail<N_JOINTS>();
+          m_q_sot.tail(m_robot_util->m_nbJoints) = q.tail(m_robot_util->m_nbJoints);
           base_se3_to_sot(m_Mff.translation()-righ_foot_sole_xyz,
                           m_Mff.rotation(),
-                          m_q_sot.head<6>());
+                          m_q_sot.head(6));
```
Comment (andreadelprete) [inline src/free-flyer-locator.cpp]: I see no reason to change that...

**Review by andreadelprete — CHANGES_REQUESTED** [review]
This is just a partial review I did up to inverse-dynamics-balance-controller.cpp (excluded)

File: src/position-controller.cpp [hunk 5]
```diff
@@ -71,20 +70,23 @@ namespace dynamicgraph
...
 
         /* Commands. */
         addCommand("init",
-                   makeCommandVoid1(*this, &PositionController::init,
-                                    docCommandVoid1("Initialize the entity.",
-                                                    "Time period in seconds (double)")));
+                   makeCommandVoid2(*this, &PositionController::init,
+                                    docCommandVoid2("Initialize the entity.",
+                                                    "Time period in seconds (double)",
+						    "Size of the state vector (double)")));
```
Comment (andreadelprete) [inline src/position-controller.cpp]: This comment is a bit ambiguous. I suggest replacing "size of the state vector" with something like "number of controlled joints".

File: unitTesting/test_se3_trajectory_generator.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+from dynamic_graph.sot.core.matrix_util import matrixToTuple, vectorToTuple,rotate, matrixToRPY
+from dynamic_graph.sot.torque_control.se3_trajectory_generator import *
+from numpy import matrix, identity, zeros, eye, array, pi, ndarray, ones
+
+# Instanciate a pose trajectory generator
+initial_value=zeros(6)
+se3tg=SE3TrajectoryGenerator("se3tg_test")
+se3tg.initial_value.value = initial_value
+se3tg.recompute(10)
```
Comment (andreadelprete) [inline unitTesting/test_se3_trajectory_generator.py]: I think you should first initialize the entity, and then call ```recompute``` on the output signal (and not directly on the entity).

File: unitTesting/unit_test_inverse_dynamics_balance_controller.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+# Com ref
+comRefPos=zeros(3)
+comRefVel=zeros(3)
+comRefAcc=zeros(3)
+idbc.com_ref_pos.value=comRefPos
+idbc.com_ref_vel.value=comRefVel
+idbc.com_ref_acc.value=comRefAcc
+
+# Right foot
+rfRefPos=zeros(12)
```
Comment (andreadelprete) [inline unitTesting/unit_test_inverse_dynamics_balance_controller.py]: The first 9 values of this vector represent a rotation matrix, so you should not set all elements to zero.

**Review by andreadelprete — CHANGES_REQUESTED** [review]
I finished the review I started yesterday!

File: src/common-py.cpp (hunk 1 above)
Comment (olivier-stasse) [reply src/common-py.cpp]: cp is for copy and avoid sending back reference

File: src/free-flyer-locator.cpp (hunk 3 above)
Comment (olivier-stasse) [reply src/free-flyer-locator.cpp:115]: It should be...

File: unitTesting/test_se3_trajectory_generator.py (hunk 6 above)
Comment (olivier-stasse) [reply unitTesting/test_se3_trajectory_generator.py]: Fix in commit 2b13647b1

File: unitTesting/unit_test_inverse_dynamics_balance_controller.py (hunk 7 above)
Comment (olivier-stasse) [reply unitTesting/unit_test_inverse_dynamics_balance_controller.py]: Fix in 2009b21a

**Pull request closed without merging**